clustered/common/src/main/java/org/ehcache/clustered/common/internal/store/Util.java

```diff
@@ -149,23 +150,27 @@ public ByteBuffer getPayload() {
     };
   }
 
-  public static Object unmarshall(byte[] payload) {
+  public static Object unmarshallWhiteListed(byte[] payload, List<Class> whiteListedClasses) {
     ObjectInputStream objectInputStream = null;
     try {
-      objectInputStream = new ObjectInputStream(new ByteArrayInputStream(payload));
+      objectInputStream = new WhiteListObjectInputStream(new ByteArrayInputStream(payload), whiteListedClasses);
       return objectInputStream.readObject();
+    } catch (InvalidClassException e) {
+      throw new RuntimeException(e.getMessage());
     } catch (IOException | ClassNotFoundException ex) {
       throw new IllegalArgumentException(ex);
     } finally {
       closeSilently(objectInputStream);
```

Comment:
We should be able to switch to try-with-resources and eliminate the `closeSilently(...)` method.  Can probably also eliminate the duplication between the `byte[]` and `ByteBuffer` variants while we're at it.

Reply:
Sure - you'll get a 'supressed' exception instead which is better in all ways.

Reply:
removed closeSilently()